src/policy/policy.cpp

```diff
@@ -34,11 +34,25 @@ CAmount GetDustThreshold(const CTxOut& txout, const CFeeRate& dustRelayFee)
     return 3 * dustRelayFee.GetFee(nSize);
 }
 
+CAmount GetDustThreshold(const CFeeRate& dustRelayFee)
+{
+    // return the dust threshold for a typical 34 bytes output
+    return 3 * dustRelayFee.GetFee(182);
+}
+
 bool IsDust(const CTxOut& txout, const CFeeRate& dustRelayFee)
 {
     return (txout.nValue < GetDustThreshold(txout, dustRelayFee));
 }
 
+CAmount GetShieldedDustThreshold(const CFeeRate& dustRelayFee)
+{
+    unsigned int K = DEFAULT_SHIELDEDTXFEE_K;   // Fixed (1000) for now
+    return 3 * K * dustRelayFee.GetFee(SPENDDESCRIPTION_SIZE +
+                                       CTXOUT_REGULAR_SIZE +
+                                       BINDINGSIG_SIZE);
```

Comment:
why adding 34 bytes more from `CTXOUT_REGULAR_SIZE` on a shielded dust threshold calculation?

Reply:
Because here we want to return a value equal to 3 times the minimum fee that would be required to spend a note.

If you want to spend a note, using the minimum possible fee, then you need to do a transaction with one shielded input, one transparent output (as it is smaller than a shielded output), and no change.